File: server/app/schemas/export.py
from typing import Literal

from pydantic import BaseModel


class PdfExportRequest(BaseModel):
    invoice_ids: list[int]
    layouts: dict[str, Literal["portrait", "landscape"]] = {}


class BatchPdfExportRequest(BaseModel):
    layouts: dict[str, Literal["portrait", "landscape"]] = {}


class ReimbursementItem(BaseModel):
    expense_item: str
    amount: float


class ReimbursementPreviewResponse(BaseModel):
    department: str
    report_date: str | None = None
    reporter: str
    items: list[ReimbursementItem]
    total_amount: float
    total_amount_cn: str

File: web/src/types/export.ts
export interface ReimbursementItem {
  expense_item: string;
  amount: number;
}

export interface ReimbursementPreview {
  department: string;
  report_date: string | null;
  reporter: string;
  items: ReimbursementItem[];
  total_amount: number;
  total_amount_cn: string;
}

export interface PdfExportRequest {
  invoice_ids: number[];
  layouts: Record<string, 'portrait' | 'landscape'>;
}

export interface BatchPdfExportRequest {
  layouts: Record<string, 'portrait' | 'landscape'>;
}